Scripts/Runtime/Data/IDProviderAtomicBurstable.cs

```diff
@@ -0,0 +1,76 @@
+using Anvil.CSharp.Data;
+using System.Threading;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Data
+{
+    /// <summary>
+    /// An <see cref="IDProvider"/> that is compatible with Burst and will get the next ID atomically so it is
+    /// safe to use in threaded context.
+    /// </summary>
+    [BurstCompatible]
+    public readonly unsafe struct IDProviderAtomicBurstable
+    {
+        public const uint DEFAULT_SUPPLY_WARNING_THRESHOLD = uint.MaxValue - 1_000_000;
+        
+        /// <summary>
+        /// The threshold to signify the ID supply is near exhaustion
+        /// </summary>
+        public readonly uint SupplyWarningThreshold;
+        
+        [NativeDisableUnsafePtrRestriction] private readonly uint* m_IDPointer;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="IDProviderAtomicBurstable"/>.
+        /// </summary>
+        /// <remarks>
+        /// NOTE: Because this is a struct, it could be created by using the default constructor which will fail.
+        /// Debug.Asserts will catch this at runtime when used.
+        /// It is expected to create this struct and pass in an explicit supply warning threshold instead, the
+        /// <see cref="DEFAULT_SUPPLY_WARNING_THRESHOLD"/> is provided to make this easy to do so.
+        ///
+        /// ALSO NOTE: Because this must be compatible with Burst, there is no event to signify when the
```

Comment:
I wonder if we should create a sibling system that does this for the developer and emits the event. Could just be a TODO

Reply:
Added a TODO to #265 
Good idea.